mk/clang.mk

```diff
@@ -19,6 +19,8 @@ ifeq ($(sm)-$(CFG_WITH_PAGER),core-y)
 # s->name) so perhaps a new command line option could be proposed upstream?
 # Anyway, use GNU.ld for the moment.
 LDcore		:= $(CROSS_COMPILE_$(sm))ld
+else
+comp-ldflags-allow-text-reloc-clang := -Wl,-z,notext
```

Comment:
Should be `-z notext` because it's linker flags, not compiler flags for the linker.

With that fixed, I get:
```
ld.lld: error: section: .ctors is not contiguous with other relro sections
ld.lld: error: section: .dtors is not contiguous with other relro sections
```

...which can be fixed by moving `.ctors` and `.dtors` just after `.plt`. Then the build goes fine but the board hangs on boot. I'll take a closer look.

Reply:
FYI, the Clang build works OK on my HiKey960 if GNU ld is used.

Reply:
QEMU virt v7 works fine with clang with the ctors/dtors fix applied.
QEMU virt v8 crashes due to stack overrun (filling translation tables with junk) on the TMP stack during initialization.

It seems it's `kprintf()` and friends whos using a lot of stack.

Reply:
Same situation with on HiKey960: 32-bit OK, 64-bit hangs during initialization (tested commit 235cf33c with clang).